syntax = "proto3";

import "empty.proto";

option go_package = "github.com/mkacz91/spejs/pb";

// Meta-service that allows interacting with a job running a gRPC server.
service JobService {
  // First request sent to the job by the launcher to retrieve runtime
  // information.
  rpc Attach (Empty) returns (JobAttachResponse) {}

  // Returns the status of the job.
  rpc Status (Empty) returns (JobStatusResponse) {}

  // Requests the job to gracefully release all resources and quit.
  //
  // Response only indicates acknowledgment, and the job will likely exit some
  // time after.
  rpc Quit (Empty) returns (Empty) {}
}

message JobAttachResponse {
  // Command used to start the job.
  string command = 1;

  // The PID of the process running the job.
  int32 pid = 2;
}

message JobStatusResponse {
  // Whether the job is ready to accept requests.
  bool is_ready = 1;
}

